cirq-core/cirq/study/symbol_func.py

```diff
@@ -41,7 +41,9 @@ def __init__(self, expr: sympy.Basic) -> None:
         self.expr = expr
         self.param_set = protocols.parameter_names(expr)
         self.params = sorted(self.param_set)
-        self.func = sympy.lambdify(self.params, expr)
+        func_args = [f'_{i}' for i in range(len(self.params))]
+        self.func_expr = expr.subs({param: arg for param, arg in zip(self.params, func_args)})
+        self.func = sympy.lambdify(func_args, self.func_expr)
```

Comment:
Do you know if the body of `func_expr` presents a risk independent of the args? i.e. could a user somehow construct an expression like `a someBadOp b`, or are all operators converted to a known set prior to calling `exec`?

Reply:
There are definitely still ways to construct bad expressions, for example you could create a `sympy.Function('print')` and this will turn into an actual call to python's `print` when lambdifying.